### PR TITLE
Add action types to action generator

### PR DESCRIPTION
**Pull request by hannah-bulmer**
Added custom action "types" via the OIH standards, also added config fields to `component.json` for `objectType` for actions that require it.

### CODE REVIEW

File: generators/action/index.js [hunk 1]
```diff
@@ -1,90 +1,104 @@
...
+const ACTIONS_WITH_OBJECTS = ['lookupById', 'delete', 'lookupMany'];
 
 module.exports = class extends Generator {
   initializing() {
     try {
-      fs.accessSync(this.destinationPath('component.json'));
+      fs.existsSync(this.destinationPath('component.json'));
       this.compDesc = this.fs.readJSON(this.destinationPath('component.json'), {});
-      // Have Yeoman greet the user.
-      this.log(yosay(
```
Comment (dvd-z) [inline generators/action/index.js]: That poor yeoman

File: generators/action/index.js [hunk 2]
```diff
@@ -1,90 +1,104 @@
...
-        {
-          name: 'Static (known at design time)',
-          short: 'Static',
-          value: 'Static',
-        },
-        {
-          name: 'Dynamic (fetched at run time)',
-          short: 'Dynamic',
-          value: 'Dynamic',
+    const prompt = prompts();
```
Comment (dvd-z) [inline generators/action/index.js]: 👍

File: generators/action/index.js [hunk 3]
```diff
@@ -99,9 +113,16 @@ module.exports = class extends Generator {
...
-      this.templatePath('action.js'),
+    const actionFile = actionType;
+    this.log(actionFile);
+
+    this.fs.copyTpl(
+      this.templatePath(`${actionFile}.js`),
       this.destinationPath(`lib/actions/${id}.js`),
+      (() => {
+        if (mType === 'Static') return { metadata: '' };
+        return { metadata: 'exports.getMetaModel = async function getMetaModel(cfg) {};' };
```
Comment (dvd-z) [inline generators/action/index.js]: will the static option leave 2 newlines at the end of templates? if so, can you delete 1 newline in the templates and change the dynamic option to return `exports.getMetaModel = ...` along with 1 newline?

File: package.json [hunk 4]
```diff
@@ -15,6 +15,7 @@
     "generator-node": "2.7.0",
     "lodash": "4.17.15",
     "mkdirp": "0.5.1",
+    "rimraf": "^3.0.0",
```
Comment (dvd-z) [inline package.json]: carat

**Review by dvd-z — CHANGES_REQUESTED** [review]

File: generators/action/index.js (hunk 2 above)
Comment (hannah-bulmer) [reply generators/action/index.js]: God bless organization

**Review by dvd-z — APPROVED** [review]

**Pull request merged**